src/injected/element-finder-by-path.ts

```diff
@@ -35,6 +35,10 @@ export class ElementFinderByPath {
     };
 
     public processRequest = (message: ElementFinderByPathMessage): PromiseLike<string> => {
+        if (!this.checkSyntax(message.path[0])) {
+            return Promise.resolve('error');
```

Comment:
I think it would be a little tough for someone seeing this message to understand why it's happening. `Promise`s support a specific mechanism (`.reject`) for reporting errors; I'd recommend using that, since that's what an async caller will generally be expecting it to work like. That also has the advantage of letting you use an `Error` instead of a `string` to represent the error message, which is nice because it captures the stack that the error occurred at and gives someone looking at the error message file/line information about where the error occurred.

Using it might look something like:

```ts
return Promise.reject(new Error(`Syntax error in selector path "${message.path[0]}"`));
```

Reply:
This makes a ton of sense^ I was following a pattern which just resolved with null upon error because there are nested promises and it made it a bit easier to tell what error was coming from what, but giving the error a significant message seems much better. I'll put this in now! Thank you (and thank you for the great explanation!)

Reply:
After talking to Wahaj we left out the Error specification message because we aren't going to give users different errors based on the situation (quite yet!)

Reply:
But resolves() calls have been transitioned to rejects()